swhkd/src/daemon.rs

```diff
@@ -143,15 +143,25 @@ async fn main() -> Result<(), Box<dyn Error>> {
         };
     }
 
-    let arg_devices: Vec<&str> = args.values_of("device").unwrap_or_default().collect();
+    let arg_add_devices: Vec<&str> = args.values_of("device").unwrap_or_default().collect();
+    let arg_ignore_devices: Vec<&str> =
+        args.values_of("ignore-device").unwrap_or_default().collect();
 
     let keyboard_devices: Vec<_> = {
-        if arg_devices.is_empty() {
+        if arg_add_devices.is_empty() {
             log::trace!("Attempting to find all keyboard file descriptors.");
-            evdev::enumerate().filter(|(_, dev)| check_device_is_keyboard(dev)).collect()
+            evdev::enumerate()
+                .filter(|(_, dev)| {
+                    !arg_ignore_devices.contains(&dev.name().unwrap_or(""))
```

Comment:
Hi, instead of repeating this same section further down in the patch, can we instead have a compile time expanded macro or a simple function call?